### PR TITLE
Replace the Serilog, Actors and metrics providers with more generic versions

### PR DESCRIPTION
**Pull request by s-vitaliy**
This pull request includes changes to various files that primarily focus on improving code readability and flexibility. The most important changes include modifying the `minor_v` version in the GitHub workflow, refactoring methods in the `ActorProviderExtensions.cs` and `DefaultLoggingProvider.cs` files for better readability and flexibility, and updating the `DatadogServiceProvider.cs` file to make the metrics service configuration more flexible.

GitHub Workflow Changes:
* [`.github/workflows/prepare-github-release.yaml`](diffhunk://#diff-8b998ec6d89a632f5ac59a469fa8d10c7e0aa8e5c10fbf553ce1c54cc26001e4L19-R19): The `minor_v` version has been incremented from 0 to 1.

Code Readability and Flexibility:
* [`src/ActorProviders/ActorProviderExtensions.cs`](diffhunk://#diff-f73d40de535d051a69e0bd6de2c48c4439abb3c8c8629c9a08879a8d62b76b32R117-R126): The method `AddLocalActorSystem` has been refactored to include an optional `configureAction` parameter for `ActorSystem` configuration. This change allows for more flexibility when adding a local actor system. [[1]](diffhunk://#diff-f73d40de535d051a69e0bd6de2c48c4439abb3c8c8629c9a08879a8d62b76b32R117-R126) [[2]](diffhunk://#diff-f73d40de535d051a69e0bd6de2c48c4439abb3c8c8629c9a08879a8d62b76b32L144-L145)
* [`src/Logs/Providers/DefaultLoggingProvider.cs`](diffhunk://#diff-b924eb4648782838fd730b70404fd8c16558767fcde30a17d935621182bf78efL22-R63): Several methods have been refactored to improve code readability and flexibility. The `AddSerilogLogger` method has been simplified, and a new method `BaseConfiguration` has been added to handle base logger configuration. The `CreateBootstrappLogger` method has also been updated to allow for optional configuration. [[1]](diffhunk://#diff-b924eb4648782838fd730b70404fd8c16558767fcde30a17d935621182bf78efL22-R63) [[2]](diffhunk://#diff-b924eb4648782838fd730b70404fd8c16558767fcde30a17d935621182bf78efR73-R87)

Metrics Service Configuration:
* [`src/Metrics/Providers/DatadogServiceProvider.cs`](diffhunk://#diff-23ec93c5df3c842fdaf5d102436849b0de771ad377380992b670f2d6752ca154L1-R44): The methods for adding Datadog metrics have been updated to include an optional `configureAction` parameter. This change makes the metrics service configuration more flexible.

### CODE REVIEW

File: src/ActorProviders/ActorProviderExtensions.cs [hunk 1]
```diff
@@ -114,34 +114,37 @@ public static ByteString Decompress(this byte[] bytes)
...
+         {
+             ConfigureActorLogging(builder);
+             configureAction?.Invoke(builder);
+         }).AddSingleton<IMaterializer>(provider => provider.GetRequiredService<ActorSystem>().Materializer());
+
+        private static void ConfigureActorLogging(AkkaConfigurationBuilder builder) =>
+            builder.ConfigureLoggers(loggerBuilder =>
             {
-                builder.ConfigureLoggers(loggerBuilder =>
+                loggerBuilder.LogLevel = Environment.GetEnvironmentVariable("PROTEUS__DEFAULT_LOG_LEVEL") switch
```
Comment (george-zubrienko) [inline src/ActorProviders/ActorProviderExtensions.cs:129]: use domain env variable

File: src/Logs/Providers/DefaultLoggingProvider.cs [hunk 2]
```diff
@@ -63,5 +70,20 @@ public static Serilog.ILogger CreateBootstrappLogger(string applicationName)
                 .Enrich.WithProperty("Environment", Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? "Local")
                 .Enrich.WithProperty("ApplicationVersion", Environment.GetEnvironmentVariable("APPLICATION_VERSION") ?? "v0.0.0");
         }
+
+        private static LoggerConfiguration BaseConfiguration(this LoggerConfiguration loggerConfiguration,
+            IServiceProvider services, string applicationName)
+        {
+            return (Environment.GetEnvironmentVariable("PROTEUS__DEFAULT_LOG_LEVEL") switch
```
Comment (george-zubrienko) [inline src/Logs/Providers/DefaultLoggingProvider.cs:77]: same here (var name would be DEFAULT_LOG_LEVEL)

**Review by george-zubrienko — APPROVED** [review]

File: src/ActorProviders/ActorProviderExtensions.cs (hunk 1 above)
Comment (s-vitaliy) [reply src/ActorProviders/ActorProviderExtensions.cs]: Fixed

File: src/Logs/Providers/DefaultLoggingProvider.cs (hunk 2 above)
Comment (s-vitaliy) [reply src/Logs/Providers/DefaultLoggingProvider.cs]: Fixed

**Pull request merged**